[Gardening] Add troubleshooting links to getting started (#60322)
swiftlang/swift: docs/HowToGuides/GettingStarted.md

   - If you use an editor other than Xcode and/or you want somewhat faster builds,
     go with Ninja.
   - If you are comfortable with using Xcode and would prefer to use it,

>      go with Xcode. But, since Xcode build is not as stable as Ninja's, you can also try to use [a ninja build integrated into Xcode](#integrate-a-ninja-build-with-xcode).

Reviewer: ```suggestion
     go with Xcode. If you run into issues building with Xcode, you can alternatively [integrate a Ninja build into Xcode](#integrate-a-ninja-build-with-xcode).
```
or
```suggestion
     go with Xcode. But, since Xcode builds are not as stable, you may want to consider [integrating a Ninja build into Xcode](#integrate-a-ninja-build-with-xcode).
```

space and (2) you need to maintain the two builds in sync, which needs extra
care when moving across branches.

### Integrate a Ninja build with Xcode

> It is even possible to integrate the Ninja build into Xcode. For details on how to set this up see [Using Ninja with Xcode in DevelopmentTips.md](/docs/DevelopmentTips.md#using-ninja-with-xcode).

Reviewer 2: I think we should drop 'even' here :)

Author: Fixed! Thanks =]

  In many situations, there are several errors, so scrolling further back
  and looking at the first error may be more helpful than simply looking
  at the last error.
- Check if others have encountered the same issue on the [Swift Forums](https://forums.swift.org/c/development/compiler) or on
   [Swift repository 'Issues' tab][Swift Issues].

>   - Here is a list of threads that describe common issues:

Reviewer: Maybe merge this into the parent bullet?

Reviewer 2: +1

> - Check if there is already a [Swift Forums] thread from someone else with the same issue.

Reviewer: The current phrasing feels more streamline to me, and we also point to GH issues. Was there something wrong with it?

Author: I was thinking in guide those sort of "getting help" with build issues to the forums which feels like a more suitable place to get actual guidance on some issue. And from there after discussion, people could create GH issues if they feel is an actual build-script or actual build issue that should be fixed. But if you think is still worth it I can add it back :)

Reviewer: Right, but we are only suggesting to *search* for issues besides forum threads, not create them.

    * [Error building the compiler (even with ninja)](https://forums.swift.org/t/error-building-the-compiler-even-with-ninja/54834)
    * [Build failure on Apple MacBook Pro with Apple M1 Chip](https://forums.swift.org/t/build-failure-on-apple-silicon-m1-mac-mini/45011)
    * [CMake cannot compile a test program](https://forums.swift.org/t/build-failure-locally/55695)
    * [Building Swift compiler from source fails when not using Ninja](https://forums.swift.org/t/building-swift-compiler-from-source-fails-when-not-using-ninja/54656)
    * [ALL_BUILD Target failing at validation](https://forums.swift.org/t/help-building-swift-in-xcode-error/49728)

>     * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

Author: People normally use `compiler` or `Developer` tag on the forums, but we already suggesting looking up there. About GH not sure, I think for getting help on those sorts of things forums would be the best place to suggest.

>     * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

Reviewer: I believe we need a more specific tag for these "cannot build the compiler" threads and issues. I am fine with a list of threads for now though. What I think we eventually ought to do is expand the troubleshooting section to include a list of the most common problems and their inline solutions, and fall back to encouraging to search by tag.

>     * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

Reviewer 3: @rjmccall @shahmishal do you know who has ability to create Swift Forum tags?

>     * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

Reviewer 4: Everyone should be able to create tags

>     * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

Author: Oh sorry, I guess I was thinking about categories 
But maybe tags should be good enough to filter compiler issues, so we would recommend for people to create new forums threads understand delopment/compiler category and with a `building compiler` tag...

    * [“gtest/gtest.h” not found while compiling the compiler](https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399)

> - If you still could not find a solution to your issue, feel free to create a new [Swift Forums] thread in the [Development/Compiler](https://forums.swift.org/c/development/compiler) category: 

Reviewer: ```suggestion
- If you still could not find a solution to your issue, feel free to create a new Swift Forums thread in the [Development/Compiler](https://forums.swift.org/c/development/compiler) category: 
```

>   - If after going through the forums you still could not found a solution for the issue you are seeing, feel free to create a new one.

Reviewer: The top-level bullet list ought to be interpreted as a sequence of actions. Do we really need to nest this one and introduce another level?

>     - When creating a new Swift forums thread in the `Development` category: 

Reviewer: This sounds like it could be merged into the parent bullet.

  In many situations, there are several errors, so scrolling further back
  and looking at the first error may be more helpful than simply looking
  at the last error.

> - Check if there is already a [Swift Forums] thread in the `compiler`/`development` category or an issue in 
>    [Swift repository 'Issues' tab][Swift Issues] from someone else with the same issue.

Reviewer: ```suggestion
- Check if there is already a [Swift Forums](https://forums.swift.org/c/development/compiler) thread or an issue on 
   [Swift repository 'Issues' tab][Swift Issues] from someone else with the same issue.
```

The current phrasing still reads better to me though.
